File: python/3.py
# The prime factors of 13195 are 5, 7, 13 and 29.
#
# What is the largest prime factor of the number 600851475143 ?
#
# https://projecteuler.net/problem=3

import string, sys, math
import functools

import commonutils

def main():
    last = 0
    lst = []
    num = 600851475143
    for i in range(2, int(math.sqrt(num))):
        if not commonutils.isPrime(i):
            continue
        elif commonutils.isPrime(i) and num%i == 0:
            lst.append(i)

    maxPrime = functools.reduce( (lambda x, y: max(x, y)), lst)
    print (maxPrime)

if __name__ == '__main__':
    main()

File: c/3.c
/* http://projecteuler.net/problem=3 */
#include <stdio.h>
#include <math.h>

int is_prime(long num) {
        long i=0;
        
        if (num == 2) {
            return 1;
        }

        for (i=2 ; i<sqrt(num) ; i++) {
                if (num%i == 0)
                        return 0;
        }

        return 1;
}

int main(int argc, char* argv[])
{
        long num = 600851475143;
        long max = 0;
        int i=0;

        for (i=2 ; i<sqrt(num) ; i++) {
                if (is_prime(i) && (num%i == 0)) {
                        max = i;
                }
        }

        printf("max = %d\n", max);

        return 0;
}
